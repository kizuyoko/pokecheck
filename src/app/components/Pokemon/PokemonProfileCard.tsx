import Card from  "../ui/Card";
import { PokemonDataTSTypes } from "../../types/pokemonDataTSTypes";
import PokemonImageSection from "./PokemonImageSection";
import PokemonNameSection from "./PokemonNameSection";
import PokemonBasicInfoSection from "./PokemonBasicInfoSection";
import PokemonStatsSection from "./PokemonStatsSection";

type Props = {
  pokemon: PokemonDataTSTypes;
};

const PokemonProfileCard = ({ pokemon }: Props) => {
  return (
    <Card className="flex flex-col md:flex-row md:items-start gap-x-12">
      <div className="items-center flex justify-center">
        <PokemonImageSection
          id={pokemon.id} 
          name={pokemon.name}
        />
      </div>
      <div className="flex flex-col">
        <PokemonNameSection 
          name={pokemon.name}
          id={pokemon.id}
        />
        <div className="pokemon-details">
          <PokemonBasicInfoSection 
            height={pokemon.height}
            weight={pokemon.weight}
            types={pokemon.types}
            gender_rate={pokemon.gender_rate}
            genera={pokemon.genera}
          />
          <PokemonStatsSection 
            abilities={pokemon.abilities}
            stats={pokemon.stats}
          />
        </div>
      </div>
    </Card>
  );
};

export default PokemonProfileCard;