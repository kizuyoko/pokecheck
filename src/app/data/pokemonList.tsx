import type { PokemonDataTS } from "../types/pokemonDataTSTypes";

const pokemonList = [
  { id: 1, name: "Bulbasaur", url: "https://pokeapi.co/api/v2/pokemon/1/", imageURL: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png" },
  { id: 2, name: "Ivysaur", url: "https://pokeapi.co/api/v2/pokemon/2/", imageURL: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/2.png" },
  { id: 3, name: "Venusaur", url: "https://pokeapi.co/api/v2/pokemon/3/", imageURL: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/3.png" },
  { id: 4, name: "Charmander", url: "https://pokeapi.co/api/v2/pokemon/4/", imageURL: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/4.png" },
  { id: 5, name: "Charmeleon", url: "https://pokeapi.co/api/v2/pokemon/5/", imageURL: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/5.png" },
  { id: 6, name: "Charizard", url: "https://pokeapi.co/api/v2/pokemon/6/", imageURL: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/6.png" },
  { id: 7, name: "Squirtle", url: "https://pokeapi.co/api/v2/pokemon/7/", imageURL: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/7.png" },
  { id: 8, name: "Wartortle", url: "https://pokeapi.co/api/v2/pokemon/8/", imageURL: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/8.png" },
  { id: 9, name: "Blastoise", url: "https://pokeapi.co/api/v2/pokemon/9/", imageURL: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/9.png" },
  { id: 10, name: "Caterpie", url: "https://pokeapi.co/api/v2/pokemon/10/", imageURL: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/10.png" },
  { id: 11, name: "Metapod", url: "https://pokeapi.co/api/v2/pokemon/11/", imageURL: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/11.png" },  
] as PokemonDataTS[];


export default pokemonList;