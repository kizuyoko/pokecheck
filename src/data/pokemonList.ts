import type { Pokemon } from "../types/pokemon";

const pokemonList = [
  {
    "id": 1,
    "name": "bulbasaur",
    "types": [
      "grass",
      "poison"
    ],
    "abilities": [
      "overgrow",
      "chlorophyll"
    ],
    "height": 7,
    "weight": 69,
    "stats": [
      {
        "stat": "hp",
        "value": 45
      },
      {
        "stat": "attack",
        "value": 49
      },
      {
        "stat": "defense",
        "value": 49
      },
      {
        "stat": "special-attack",
        "value": 65
      },
      {
        "stat": "special-defense",
        "value": 65
      },
      {
        "stat": "speed",
        "value": 45
      }
    ],
    "gender_rate": 1,
    "genera": [
      "Seed Pokémon"
    ]
  },
  {
    "id": 2,
    "name": "ivysaur",
    "types": [        
      "grass",        
      "poison"
    ],
    "abilities": [
      "overgrow",
      "chlorophyll"
    ],
    "height": 10,
    "weight": 130,
    "stats": [
      {
        "stat": "hp",
        "value": 60
      },
      {
        "stat": "attack",
        "value": 62
      },
      {
        "stat": "defense",
        "value": 63
      },
      {
        "stat": "special-attack",
        "value": 80
      },
      {
        "stat": "special-defense",
        "value": 80
      },
      {
        "stat": "speed",
        "value": 60
      }
    ],
    "gender_rate": 1,
    "genera": [
      "Seed Pokémon"
    ]
  },  
  {
    "id": 3,
    "name": "venusaur",
    "types": [
      "grass",
      "poison"
    ],
    "abilities": [
      "overgrow",
      "chlorophyll"
    ],
    "height": 20,
    "weight": 1000,
    "stats": [
      {
        "stat": "hp",
        "value": 80
      },
      {
        "stat": "attack",
        "value": 82
      },
      {
        "stat": "defense",
        "value": 83
      },
      {
        "stat": "special-attack",
        "value": 100
      },
      {
        "stat": "special-defense",
        "value": 100
      },
      {
        "stat": "speed",
        "value": 80
      }
    ],
    "gender_rate": 1,
    "genera": [
      "Seed Pokémon"
    ]
  },
  {
    "id": 4,
    "name": "charmander",
    "types": [
      "fire"
    ],
    "abilities": [
      "blaze",
      "solar-power"
    ],
    "height": 6,
    "weight": 85,
    "stats": [
      {
        "stat": "hp",
        "value": 39
      },
      {
        "stat": "attack",
        "value": 52
      },
      {
        "stat": "defense",
        "value": 43
      },
      {
        "stat": "special-attack",
        "value": 60
      },
      {
        "stat": "special-defense",
        "value": 50
      },
      {
        "stat": "speed",
        "value": 65
      }
    ],
    "gender_rate": 1,
    "genera": [
      "Lizard Pokémon"
    ]
  },
  {
    "id": 5,
    "name": "charmeleon",
    "types": [
      "fire"
    ],
    "abilities": [
      "blaze",
      "solar-power"
    ],
    "height": 11,
    "weight": 190,
    "stats": [
      {
        "stat": "hp",
        "value": 58
      },
      {
        "stat": "attack",
        "value": 64
      },
      {
        "stat": "defense",
        "value": 58
      },
      {
        "stat": "special-attack",
        "value": 80
      },
      {
        "stat": "special-defense",
        "value": 65
      },
      {
        "stat": "speed",
        "value": 80
      }
    ],
    "gender_rate": 1,
    "genera": [
      "Flame Pokémon"
    ]
  },
  {
    "id": 6,
    "name": "charizard",
    "types": [
      "fire",
      "flying"
    ],
    "abilities": [
      "blaze",
      "solar-power"
    ],
    "height": 17,
    "weight": 905,
    "stats": [
      {
        "stat": "hp",
        "value": 78
      },
      {
        "stat": "attack",
        "value": 84
      },
      {
        "stat": "defense",
        "value": 78
      },
      {
        "stat": "special-attack",
        "value": 109
      },
      {
        "stat": "special-defense",
        "value": 85
      },
      {
        "stat": "speed",
        "value": 100
      }
    ],
    "gender_rate": 1,
    "genera": [
      "Flame Pokémon"
    ]
  },
  {
    "id": 7,
    "name": "squirtle",
    "types": [
      "water"
    ],
    "abilities": [
      "torrent",
      "rain-dish"
    ],
    "height": 5,
    "weight": 90,
    "stats": [
      {
        "stat": "hp",
        "value": 44
      },
      {
        "stat": "attack",
        "value": 48
      },
      {
        "stat": "defense",
        "value": 65
      },
      {
        "stat": "special-attack",
        "value": 50
      },
      {
        "stat": "special-defense",
        "value": 64
      },
      {
        "stat": "speed",
        "value": 43
      }
    ],
    "gender_rate": 1,
    "genera": [
      "Tiny Turtle Pokémon"
    ]
  },
  {
    "id": 8,
    "name": "wartortle",
    "types": [
      "water"
    ],
    "abilities": [
      "torrent",
      "rain-dish"
    ],
    "height": 10,
    "weight": 225,
    "stats": [
      {
        "stat": "hp",
        "value": 59
      },
      {
        "stat": "attack",
        "value": 63
      },
      {
        "stat": "defense",
        "value": 80
      },
      {
        "stat": "special-attack",
        "value": 65
      },
      {
        "stat": "special-defense",
        "value": 80
      },
      {
        "stat": "speed",
        "value": 58
      }
    ],
    "gender_rate": 1,
    "genera": [
      "Turtle Pokémon"
    ]
  },
  {
    "id": 9,
    "name": "blastoise",
    "types": [
      "water"
    ],
    "abilities": [
      "torrent",
      "rain-dish"
    ],
    "height": 16,
    "weight": 855,
    "stats": [
      {
        "stat": "hp",
        "value": 79
      },
      {
        "stat": "attack",
        "value": 83
      },
      {
        "stat": "defense",
        "value": 100
      },
      {
        "stat": "special-attack",
        "value": 85
      },
      {
        "stat": "special-defense",
        "value": 105
      },
      {
        "stat": "speed",
        "value": 78
      }
    ],
    "gender_rate": 1,
    "genera": [
      "Shellfish Pokémon"
    ]
  },
  {
    "id": 10,
    "name": "caterpie",
    "types": [
      "bug"
    ],
    "abilities": [
      "shield-dust",
      "run-away"
    ],
    "height": 3,
    "weight": 29,
    "stats": [
      {
        "stat": "hp",
        "value": 45
      },
      {
        "stat": "attack",
        "value": 30
      },
      {
        "stat": "defense",
        "value": 35
      },
      {
        "stat": "special-attack",
        "value": 20
      },
      {
        "stat": "special-defense",
        "value": 20
      },
      {
        "stat": "speed",
        "value": 45
      }
    ],
    "gender_rate": 4,
    "genera": [
      "Worm Pokémon"
    ]
  },
  {
    "id": 11,
    "name": "metapod",
    "types": [
      "bug"
    ],
    "abilities": [
      "shed-skin"
    ],
    "height": 7,
    "weight": 99,
    "stats": [
      {
        "stat": "hp",
        "value": 50
      },
      {
        "stat": "attack",
        "value": 20
      },
      {
        "stat": "defense",
        "value": 55
      },
      {
        "stat": "special-attack",
        "value": 25
      },
      {
        "stat": "special-defense",
        "value": 25
      },
      {
        "stat": "speed",
        "value": 30
      }
    ],
    "gender_rate": 4,
    "genera": [
      "Cocoon Pokémon"
    ]
  },
  {
    "id": 12,
    "name": "butterfree",
    "types": [
      "bug",
      "flying"
    ],
    "abilities": [
      "compound-eyes",
      "tinted-lens"
    ],
    "height": 11,
    "weight": 320,
    "stats": [
      {
        "stat": "hp",
        "value": 60
      },
      {
        "stat": "attack",
        "value": 45
      },
      {
        "stat": "defense",
        "value": 50
      },
      {
        "stat": "special-attack",
        "value": 90
      },
      {
        "stat": "special-defense",
        "value": 80
      },
      {
        "stat": "speed",
        "value": 70
      }
    ],
    "gender_rate": 4,
    "genera": [
      "Butterfly Pokémon"
    ]
  },
  {
    "id": 13,
    "name": "weedle",
    "types": [
      "bug",
      "poison"
    ],
    "abilities": [
      "shield-dust",
      "run-away"
    ],
    "height": 3,
    "weight": 32,
    "stats": [
      {
        "stat": "hp",
        "value": 40
      },
      {
        "stat": "attack",
        "value": 35
      },
      {
        "stat": "defense",
        "value": 30
      },
      {
        "stat": "special-attack",
        "value": 20
      },
      {
        "stat": "special-defense",
        "value": 20
      },
      {
        "stat": "speed",
        "value": 50
      }
    ],
    "gender_rate": 4,
    "genera": [
      "Hairy Bug Pokémon"
    ]
  },
  {
    "id": 14,
    "name": "kakuna",
    "types": [
      "bug",
      "poison"
    ],
    "abilities": [
      "shed-skin"
    ],
    "height": 6,
    "weight": 100,
    "stats": [
      {
        "stat": "hp",
        "value": 45
      },
      {
        "stat": "attack",
        "value": 25
      },
      {
        "stat": "defense",
        "value": 50
      },
      {
        "stat": "special-attack",
        "value": 25
      },
      {
        "stat": "special-defense",
        "value": 25
      },
      {
        "stat": "speed",
        "value": 35
      }
    ],
    "gender_rate": 4,
    "genera": [
      "Cocoon Pokémon"
    ]
  },
  {
    "id": 15,
    "name": "beedrill",
    "types": [
      "bug",
      "poison"
    ],
    "abilities": [
      "swarm",
      "sniper"
    ],
    "height": 10,
    "weight": 295,
    "stats": [
      {
        "stat": "hp",
        "value": 65
      },
      {
        "stat": "attack",
        "value": 90
      },
      {
        "stat": "defense",
        "value": 40
      },
      {
        "stat": "special-attack",
        "value": 45
      },
      {
        "stat": "special-defense",
        "value": 80
      },
      {
        "stat": "speed",
        "value": 75
      }
    ],
    "gender_rate": 4,
    "genera": [
      "Poison Bee Pokémon"
    ]
  }

] as Pokemon[];


export default pokemonList;