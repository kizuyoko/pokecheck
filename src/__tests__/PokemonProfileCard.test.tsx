import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'; 
import PokemonProfileCard from '@/app/components/Pokemon/PokemonProfileCard';
import pokemonList from '@/data/pokemonList';

const dummyPokemon = pokemonList[0];

jest.mock('next/image', () => ({
  __esModule: true,
  default: (props: React.ImgHTMLAttributes<HTMLImageElement>) => {
    // eslint-disable-next-line @next/next/no-img-element
    return <img alt={props.alt || 'Image'} {...props} />;
  },
}));  

describe('PokemonProfileCard', () => {
  it('renders a Pokémon name', () => {
    render(<PokemonProfileCard pokemon={dummyPokemon} />);
    expect(screen.getByRole('heading', { name: /bulbasaur/i })).toBeInTheDocument();
  });

  it('renders a Pokémon ID', () => {
    render(<PokemonProfileCard pokemon={dummyPokemon} />);
    expect(screen.getByText(/#0001/i)).toBeInTheDocument();
  });

  it('renders a Pokémon image', () => {
    render(<PokemonProfileCard pokemon={dummyPokemon} />);
    const image = screen.getByAltText(/bulbasaur/i);
    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute('src', 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png');
  });

  it('renders the image with correct dimensions', () => {
    render(<PokemonProfileCard pokemon={dummyPokemon} />);
    const image = screen.getByAltText(/bulbasaur/i);
    expect(image).toHaveAttribute('width', '343');
    expect(image).toHaveAttribute('height', '343');
  });
  
  it ('renders Pokémon height and weight', () => {
    render(<PokemonProfileCard pokemon={dummyPokemon} />);  
    expect(screen.getByText(/0.7/i)).toBeInTheDocument();
    expect(screen.getByText(/6.9/i)).toBeInTheDocument();
  });

  it('renders Pokémon types', () => {
    render(<PokemonProfileCard pokemon={dummyPokemon} />);
    expect(screen.getByText(/grass/i)).toBeInTheDocument();
    expect(screen.getByText(/poison/i)).toBeInTheDocument();
  });

  it ('renders Pokémon genders', () => {
    render(<PokemonProfileCard pokemon={dummyPokemon} />);
    expect(screen.getByText(/male/i)).toBeInTheDocument();
    expect(screen.getByText(/female/i)).toBeInTheDocument();
  });

  it ('renders Pokémon Category', () => {
    render(<PokemonProfileCard pokemon={dummyPokemon} />);
    expect(screen.getByText(/seed/i)).toBeInTheDocument();
  });  

  it ('renders Pokémon abilities', () => {  
    render(<PokemonProfileCard pokemon={dummyPokemon} />);
    expect(screen.getByText(/overgrow/i)).toBeInTheDocument();
    expect(screen.getByText(/chlorophyll/i)).toBeInTheDocument();
  }
  );

  it('renders Pokémon stats', () => {
    render(<PokemonProfileCard pokemon={dummyPokemon} />);
    expect(screen.getByText(/hp/i)).toBeInTheDocument();
    expect(screen.getByText(/defense/i)).toBeInTheDocument();
  });
});