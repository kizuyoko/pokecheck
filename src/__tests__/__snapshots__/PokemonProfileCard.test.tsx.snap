// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonProfileCard matches the snapshot 1`] = `
<DocumentFragment>
  <article
    class="w-full card text-left flex flex-col md:flex-row md:items-start gap-x-12"
  >
    <div
      class="items-center flex justify-center"
    >
      <figure>
        <img
          alt="bulbasaur"
          height="343"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png"
          width="343"
        />
        <figcaption
          class="sr-only"
        >
          bulbasaur
        </figcaption>
      </figure>
    </div>
    <div
      class="flex flex-col"
    >
      <h2
        class="sm:pt-6"
      >
        Bulbasaur
      </h2>
      <p
        class="text-gray-500 dark:text-gray-200"
      >
        #0001
      </p>
      <div
        class="pokemon-details"
      >
        <div>
          <div
            class="pokemon-property"
          >
            <h3>
              <span>
                Height: 
              </span>
              <strong>
                0.7 meters
              </strong>
            </h3>
          </div>
          <div
            class="pokemon-property"
          >
            <h3>
              <span>
                Weight: 
              </span>
              <strong>
                6.9 kg
              </strong>
            </h3>
          </div>
          <div
            class="pokemon-property flex"
          >
            <h3
              class="pr-2"
            >
              <span>
                Types: 
              </span>
            </h3>
            <div
              class="flex gap-4 mt-2"
            >
              <div
                class="gap-1 flex flex-col items-center text-center"
              >
                <a
                  class="transform transition-transform duration-300 hover:scale-130"
                  href="/type/grass"
                >
                  <img
                    alt="grass"
                    height="32"
                    loading="lazy"
                    src="/type_icons/grass.svg"
                    width="32"
                  />
                </a>
                <p>
                  grass
                </p>
              </div>
              <div
                class="gap-1 flex flex-col items-center text-center"
              >
                <a
                  class="transform transition-transform duration-300 hover:scale-130"
                  href="/type/poison"
                >
                  <img
                    alt="poison"
                    height="32"
                    loading="lazy"
                    src="/type_icons/poison.svg"
                    width="32"
                  />
                </a>
                <p>
                  poison
                </p>
              </div>
            </div>
          </div>
          <div
            class="pokemon-property"
          >
            <h3>
              <span>
                Gender: 
              </span>
              <strong>
                Male, Female
              </strong>
            </h3>
          </div>
          <div
            class="pokemon-property"
          >
            <h3>
              <span>
                Category: 
              </span>
              <strong>
                Seed Pok√©mon
              </strong>
            </h3>
          </div>
        </div>
        <div>
          <div
            class="pokemon-property"
          >
            <h3>
              Abilities:
            </h3>
            <ul
              class="list-disc list-inside"
            >
              <li
                class="pl-2 text-lg sm:text-xl"
              >
                overgrow
              </li>
              <li
                class="pl-2 text-lg sm:text-xl"
              >
                chlorophyll
              </li>
            </ul>
          </div>
          <div
            class="pokemon-property"
          >
            <h3>
              Stats
            </h3>
            <ul
              class="list-disc list-inside"
            >
              <li
                class="pl-2 text-lg sm:text-xl"
              >
                hp: 45
              </li>
              <li
                class="pl-2 text-lg sm:text-xl"
              >
                attack: 49
              </li>
              <li
                class="pl-2 text-lg sm:text-xl"
              >
                defense: 49
              </li>
              <li
                class="pl-2 text-lg sm:text-xl"
              >
                special-attack: 65
              </li>
              <li
                class="pl-2 text-lg sm:text-xl"
              >
                special-defense: 65
              </li>
              <li
                class="pl-2 text-lg sm:text-xl"
              >
                speed: 45
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </article>
</DocumentFragment>
`;
